zapisy/apps/statistics/assets/components/StatisticsList.vue

```diff
@@ -0,0 +1,136 @@
+<script lang="ts">
+import Vue from "vue";
+import { mapGetters } from "vuex";
+import { CourseInfo } from "../store/courses";
+import Component from "vue-class-component";
+
+@Component({
+  computed: {
+    ...mapGetters("courses", {
+      courses: "courses",
+    }),
+    ...mapGetters("filters", {
+      tester: "visible",
+    }),
+    ...mapGetters("sorting", {
+      compare: "compare",
+    }),
+  },
+})
+export default class StatisticsList extends Vue {
+  // The list should be initialised to contain all the courses and then apply
+  // filters and sorting whenever they update.
+  coursesList: CourseInfo[] = [];
+  visibleCourses: boolean[] = [];
+
+  courses!: CourseInfo[];
+  tester!: (_: CourseInfo) => boolean;
+  compare!: (a: CourseInfo, b: CourseInfo) => number;
+
+  created() {
+    this.$store.dispatch("courses/initFromJSONTag");
+  }
+  mounted() {
+    for (let i = 0; i < this.courses.length; i++) {
+      this.visibleCourses[i] = true;
+    }
+    this.courses.sort(this.compare);
+
+    this.$store.subscribe((mutation, state) => {
+      switch (mutation.type) {
+        case "filters/registerFilter":
+          this.coursesList = this.courses.filter(this.tester);
+          this.courses.sort(this.compare);
+          for (let i = 0; i < this.courses.length; i++) {
+            this.visibleCourses[i] = false;
+          }
+          for (let i = 0; i < this.coursesList.length; i++) {
+            this.visibleCourses[
+              this.coursesList[i].alphabetical_sorting_index
+            ] = true;
+          }
+          break;
+        case "sorting/changeSorting":
+          this.coursesList = this.courses.filter(this.tester);
+          this.courses.sort(this.compare);
+          for (let i = 0; i < this.courses.length; i++) {
+            this.visibleCourses[i] = false;
+          }
+          for (let i = 0; i < this.coursesList.length; i++) {
+            this.visibleCourses[
+              this.coursesList[i].alphabetical_sorting_index
+            ] = true;
+          }
+          break;
+      }
+    });
+  }
+}
+</script>
+
+<template>
+  <table class="table table-striped">
+    <thead class="text-muted">
+      <tr>
+        <th scope="col">Prowadzący</th>
+        <th scope="col">Typ</th>
+        <th scope="col">Miejsca</th>
+        <th scope="col">Zapisani</th>
+        <th scope="col">Kolejka</th>
+        <th scope="col">Przypięci</th>
+        <th></th>
+      </tr>
+    </thead>
+    <tbody>
+      <template v-for="course in courses">
+        <tr v-show="visibleCourses[course.alphabetical_sorting_index]">
```

Comment:
```suggestion
        <tr v-show="tester(course)">
```
Całe to `coursesList` i `visibleCourses` można wywalić.

Reply:
Z jakiegoś powodu działając na samym courses bez coursesList nie ma reakcji na zmianę filtrów lub sortowanie.

Reply:
"Note that v-show doesn’t support the <template> element"
https://v2.vuejs.org/v2/guide/conditional.html

Reply:
Już nie pamiętam, jak to testowałem. Chyba stworzyłem sobie osobny komponent na te parę wierszy w tabelce odpowiadających kursowi (ale wątpię, by to było konieczne).

W każdym razie na pewno to jakoś działało (i to szybko).